packages/core/core/src/RequestTracker.js

```diff
@@ -332,18 +338,19 @@ export class RequestGraph extends ContentGraph<
     );
   }
 
-  invalidateNode(nodeId: NodeId, reason: InvalidateReason) {
+  invalidateNode(nodeId: NodeId, reason: InvalidateReason, cause?: NodeId) {
     let node = nullthrows(this.getNode(nodeId));
     invariant(node.type === REQUEST);
     node.invalidateReason |= reason;
     this.invalidNodeIds.add(nodeId);
+    this.invalidationReport.push({invalidated: nodeId, cause: cause});
```

Comment:
Can we put this behind the `process.env.PARCEL_GENERATE_INVALIDATION_REPORT` flag as well so we don't use the extra memory when it's not needed.

Reply:
Good point! fixed!